### PR TITLE
Implemented polling method

### PR DESCRIPTION
**Pull request by giacomocerquone**
Feature #3 

Ok, so here I created two function (instead of one) to partialize even more the code.
So, in this way, the user can choose to launch a one shot filter or iterate that filter "tot" times at "tot" interval.
The object parameter used to filter the response is very intuitive. It has the same structure of a message object but the values of the keys are the actual value to filter for.
Anyway the method doesn't return just the first mail but all the mails that match the given filters.

For the polling method I needed to use a promise because I couldn't find a way to make it work with async/await (actually I think there is no way to make it work without creating a promise due to the setTimeout callback nature, see http://stackoverflow.com/questions/33289726/combination-of-async-function-await-settimeout).
For the setInterval, I preferred it against chained settimeout for this reason: http://stackoverflow.com/questions/729921/settimeout-or-setinterval

hope you'll like it :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,3 +106,43 @@ account.domains().then((domains) {
   '@walkmail.ru'
 ]
 ```
+
+**Filter messages**
+This function calls automatically getMail and filter the response object based on a parameter object containing the filters.
+It returns **all** the email that will pass the filters.
+```javascript
+var filterOptions = {
```
Comment (mileszim) [inline README.md]: using `let` over `var` for object assignment is preferred

File: README.md [hunk 2]
```diff
@@ -107,3 +106,43 @@ account.domains().then((domains) {
...
+  'from': 'somemail@gmail.com',
+  'text': 'May the force be with you.'
+};
+account.filterMessages(filterOptions).then((filteredMails) => {
+  console.log(filteredMails);
+});
+
+var filterOptionsTwo = {
```
Comment (mileszim) [inline README.md]: using `let` over `var` for object assignment is preferred

File: README.md [hunk 3]
```diff
@@ -107,3 +106,43 @@ account.domains().then((domains) {
...
+**Polling the getMail response object**
+This function is basically a setInterval wrapper that constantly calls the previous filterMessages() method.
+It returns **all** the email that will pass the filters.
+The parameters are:
+  - filterOptions: Object containing the filters
+  - interval: The setInterval's interval in milliseconds
+  - iterations: Maximum number of attempt after which it'll return "Timeout exceeded"
+
+```javascript
+var filterOptions = {
```
Comment (mileszim) [inline README.md]: using `let` over `var` for object assignment is preferred

File: src/index.js [hunk 4]
```diff
@@ -35,6 +35,54 @@ export default class TempMail {
   }
 
   /**
+   * Polling for an email
+   */
+    pollingMail(filterOptions, interval = 5000, iterations = 10) {
+      return new Promise( (resolve, reject) => {
```
Comment (mileszim) [inline src/index.js]: `return new Promise((resolve, reject) => {`

File: src/index.js [hunk 5]
```diff
@@ -35,6 +35,54 @@ export default class TempMail {
...
+            reject(error);
+          }
+        }, interval);
+      });
+    }
+
+   /**
+    * Filter messages based on checks over the object's properties
+    */
+    async filterMessages(filterOptions) {
```
Comment (mileszim) [inline src/index.js]: Any way to tidy up this method?

**Review by mileszim — CHANGES_REQUESTED** [review]
Clean up your methods and compile into dist once finished. Thanks for the contribution!

**Comment by giacomocerquone** [comment]
You can just edit and add commits inside this PR instead of deleting and rewriting the code in another PR like you did!
I'm busy at the moment and I can't work on this.

**Comment by mileszim** [comment]
Closing in favor of https://github.com/mileszim/tempmail.js/blob/master/src/index.js#L76-L89

**Pull request closed without merging**